forgerock-auth/src/main/java/org/forgerock/android/auth/TokenManager.java

```diff
@@ -88,5 +91,13 @@ public interface TokenManager {
      */
     void revoke(FRListener<Void> listener);
 
+    /**
+     * OAuth2 Token Revocation, and end the session
+     *
+     * @param appAuthConfiguration The AppAuthConfiguration
+     * @param listener             Listener to listen for token revocation event.
+     */
+    void revokeAndEndSession(Supplier<AppAuthConfiguration> appAuthConfiguration, FRListener<Void> listener);
```

Comment:
Should we make second parameter optional, to avoid the need to pass null?

Reply:
For now the SessionManager.close() ignores the result of success or failure, tries best to close the session.
Would leave it to take a listener for now, we may want to receive result in the future.